### ISSUE TITLE
x96max+ a100 写入 emmc 后变砖无法启动

### ISSUE THREAD

**Issue by dicsonpan** [ophub/amlogic-s9xxx-armbian#779]
**Device Information | 设备信息**
 - SOC: [S095x3]
 - Model [x96max+ a100，无线网卡是AM7256，有线100M 那款]

**Armbian Version | 系统版本**
 - Kernel Version: [6.0.11]
 - Release: [jammy]

**Describe the bug | 问题描述**
用2022年6月份的版本，具体记不清了，也是jammy，内核5.15把，已经刷机成功，并正常写入 emmc ，也是一直能够正常运行
最近想折腾一下桌面环境，自己装老是有各种报错，非常苦恼。看了现在最新的版本，有 armbian-software 命令，可以比较方便地一键安装桌面环境，故想要重新刷机

**Comment by dicsonpan** [comment]
按照上述操作，修改了 uEnv.txt，启动进入 TF 卡上的 armbian 并运行 armbian-install，选择 517 后写入 emmc 后重启，屏幕和之前变砖不同，虽然还是全黑，但 led 屏幕的背光是亮的，即不能显示任何信息。也不能正常启动 emmc 上的 armbian
拔电源，捅菊花，usb 读卡器加载 TF 卡，还能重新进入卡上的系统

**Comment by ophub** [comment]
安装时用：armbian-install -m yes

**Comment by dicsonpan** [comment]
情况还是一样，我把 armbian-install 的主要运行过程截图下来了，看看有没什么遗漏或者出错的地方？
另外更正一下，不用捅菊花，插入 usb 口也能加载 TF 卡上的 armbian，是不是 u-boot 或者 dtb 没搞对？
![52711670388539_ pic](https://user-images.githubusercontent.com/1907869/206091141-68cd9e97-76a8-4c92-b6a2-7ee38e7dca9c.jpg)

**Comment by ophub** [comment]
```
cat /boot/uEnv.txt
sha256sum /boot/u-boot*
```

**Comment by dicsonpan** [comment]
我记不起来用哪个 dtb 了
5.15 也试过，一样

**Comment by dicsonpan** [comment]
我记不清之前用的哪个 dtb 了，那几个 dtb 我都试过了，还是一样
刷到 emmc 里边的时候，我看到有个提示说把 u-boot-x96maxplus.bin copy 为 u-boot.emmc 对吗？

**Comment by dicsonpan** [comment]
![image](https://user-images.githubusercontent.com/1907869/206098516-05e85f01-3c59-4d21-9fb1-ea4dd2084aa6.png)
5.15 固件也试过，一样的情况

**Comment by ophub** [comment]
你就改这个dtb测试

**Comment by ophub** [comment]
内存少了就复制u-boot重启

u-boot-x96maxplus.bin copy 为 u-boot.emmc

**Comment by ophub** [comment]
你的盒子拍点照片给大家参考下，
外观
底部的标签
如何方便拍主板的话，正反两面拍一下
如果知道救砖的短接点的话，在主板上标示一下
如果有它的安卓系统的话分享一下，我转存一份备用

**Comment by dicsonpan** [comment]
盒子已经没了，我一直用主板。照片稍后整理发出
现在的情况是这样
内核：5.15
dtb：meson-sm1-sei610.dtb
没有 u-boot.emmc ，能正常通过 emmc 进入系统，但内存只检测到 1G
如果把 u-boot-x96maxplus.bin 复制为 u-boot.emmc，那还是不能通过 emmc 进入系统

我是用 u盘 启动，进入 u盘 自己的 armbian 之后，把 mmcblk2p1 mount 到一个目录下，进行的 copy 操作，没有重新运行 armbian-install 命令

**Comment by ophub** [comment]
u盘是u-boot.ext
我给配置添加一条记录，你稍等下载使用这个文件

**Comment by ophub** [comment]
```
# 下面的命令一行复制到USB启动系统里执行，替换本地的配置文件
wget https://raw.githubusercontent.com/ophub/amlogic-s9xxx-armbian/main/build-armbian/armbian-files/common-files/etc/model_database.conf -O /etc/model_database.conf

# 然后再执行安装选522即可
armbian-install -m yes
```

**Comment by dicsonpan** [comment]
> ```
> # 下面的命令一行复制到USB启动系统里执行，替换本地的配置文件
> wget https://raw.githubusercontent.com/ophub/amlogic-s9xxx-armbian/main/build-armbian/armbian-files/common-files/etc/model_database.conf -O /etc/model_database.conf
> 
> # 然后再执行安装选522即可
> armbian-install -m yes
> ```

还是之前说的那个问题，emmc 不能启动 armbian。把 u-boot.ext 删除后，能进入 armbian，但内存只识别出 1G

**Comment by ophub** [comment]
在usb里把u-boot-x96maxplus.bin 复制为 u-boot.ext可以启动？
你是复制的还是改名的？只能复制不能改名，不要把u-boot-x96maxplus.bin整没了，不然写入emmc时缺少文件无法复制。

**Comment by dicsonpan** [comment]
> 在usb里把u-boot-x96maxplus.bin 复制为 u-boot.ext可以启动？这时内存是正常的吧。 
可以启动，并且内存正常显示是 4G

你是复制的还是改名的？只能复制不能改名，不要把u-boot-x96maxplus.bin整没了，不然写入emmc时缺少文件无法复制。
我都是复制的，没有改名

**Comment by ophub** [comment]
https://github.com/ophub/amlogic-s9xxx-armbian/issues/283

这个帖子里使用5.10内核。这个肯定需要u-boot才能启动，他存储32G不知道他使用的是emmc还是tf卡。

按理说tf能用的u-boot，在emmc里也没问题，你这是第一个反应只能usb使用的现象。

**Comment by ophub** [comment]
你会救砖的话，可以试试添加 `MAINLINE_UBOOT`

刚才给你添加的`522`编号这一行，第六列里把`NA`改为`x96maxplus-u-boot.bin.sd.bin`

写入时还是使用命令：`armbian-install -m yes`

**Comment by dicsonpan** [comment]
这样就可以了！

**Comment by ophub** [comment]
添加MAINLINE_UBOOT可以了？

**Comment by ophub** [comment]
把你的emmc里的/etc/ophub-release这个文件改下
MAINLINE_UBOOT='/usr/u-boot/x96maxplus-u-boot.bin.sd.bin'

这样以后更新内核时自动给添加。

**Comment by ophub** [comment]
![Snip20221207_1](https://user-images.githubusercontent.com/68696949/206135852-42524cfa-8590-400c-9ad9-b776663782c7.png)

你分享的固件里，你使用过的是哪个？下面的atv这个？

**Comment by dicsonpan** [comment]
> ![Snip20221207_1](https://user-images.githubusercontent.com/68696949/206135852-42524cfa-8590-400c-9ad9-b776663782c7.png)
> 
> 你分享的固件里，你使用过的是哪个？下面的atv这个？
是的

**Comment by ophub** [comment]
https://github.com/ophub/kernel/releases/tag/tools

安卓固件转存到这里了。
你试试5.10的应该可以，6.0的好几个盒子反馈不能用。

**Comment by luucasmooreira** [comment]
Can anyone help me with the correct procedure? I booted the firm with sei610.dtb, copied u-boot-x96maxplus.bin to u-boot.ext, the system loaded and took the 4 GB of RAM and LAN IP. I gave the following command armbian-install -m yes, and selected dtb 522. But when I restarted I didn't get IP =/.


Where am I going wrong?

**Comment by ophub** [comment]
`cat /etc/model_database.conf | grep ^522`

Aren't you asking about the s905w next door? This post is s905a, did you change the device?

**Comment by luucasmooreira** [comment]
![image](https://user-images.githubusercontent.com/118861204/206352117-c0c2853d-d439-4a6f-8649-488f72b0e60d.jpeg)



Yes, but it's the exact same problem I'm facing. I can't make IP Lan + 4GB of Ram work.

**Comment by luucasmooreira** [comment]
![image](https://user-images.githubusercontent.com/118861204/206352517-a4663c0c-bdbb-41bd-9d6d-0bb8a685d152.jpeg)

![image](https://user-images.githubusercontent.com/118861204/206352561-52ee7870-d086-4b4a-8ebb-74039a877bf9.jpeg)


Yes, but it's the exact same problem I'm facing. I can't make IP Lan + 4GB of Ram work.

**Comment by ophub** [comment]
Your 100W can be used normally?

**Comment by ophub** [comment]
From the motherboard picture you sent, your bluetooth/WIFI chip is different, his is AM7256, yours is W522A

**Comment by luucasmooreira** [comment]
yes, sei610.dtb recognizes 1gb of ram and takes ip Lan. a95xf3.air.dtb recognizes 2 GB of Ram and Ip Lan. I don't remember if x96-air.dtb worked with 2 GB Ram too, and it got Ip Lan. I've been trying to solve it for 5 days, I'm not thinking straight anymore.

Obter o Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: ophub ***@***.***>
Enviado: Thursday, December 8, 2022 12:02:29 AM
Para: ophub/amlogic-s9xxx-armbian ***@***.***>
Cc: luucasmooreira ***@***.***>; Comment ***@***.***>
Assunto: Re: [ophub/amlogic-s9xxx-armbian] x96max+ a100 支持写入emmc使用 (Issue #779)


From the motherboard picture you sent, your bluetooth/WIFI chip is different, his is AM7256, yours is W522A

―

**Comment by luucasmooreira** [comment]
yes, sei610.dtb recognizes 1gb of ram and takes ip Lan. a95xf3.air.dtb recognizes 2 GB of Ram and Ip Lan. I don't remember if x96-air.dtb worked with 2 GB Ram too, and it got Ip Lan. I've been trying to solve it for 5 days, I'm not thinking straight anymore.

**Comment by D-Sta** [comment]
Hi,
I have the same hardware: 
    SOC: [S095x3]
    Model [x96max+ a100，无线网卡是AM7256，有线100M 那款]

I have installed the armbian to the emmc with `armbian-install -m yes` with updated `/etc/model_database.conf`
and the target is 522. 
4GB of ram is recognized but the Ethernet is not longer working.

**Comment by ophub** [comment]
try 5.15/5.10 kernel

**Comment by D-Sta** [comment]
Sorry i forgot mentioning the kernel.

`/etc/ophub-release`

`PLATFORM='amlogic'
VERSION_CODEID='debian'
VERSION_CODENAME='bullseye'
SOC='s905x3'
MODEL_NAME='X96-Max+_100Mb'
FDTFILE='meson-sm1-sei610.dtb'
UBOOT_OVERLOAD='u-boot-x96maxplus.bin'
MAINLINE_UBOOT='/usr/lib/u-boot/x96maxplus-u-boot.bin.sd.bin'
BOOTLOADER_IMG='/usr/lib/u-boot/'
FAMILY='meson-sm1'
BOARD='s905x3'
KERNEL_VERSION='5.15.81'
KERNEL_BRANCH='stable'
K510='0'
ROOTFS_TYPE='btrfs'
BOOT_CONF='uEnv.txt'
DISK_TYPE='emmc'
MLUBOOT_STATUS='yes'
AMPART_STATUS='yes'
PACKAGED_DATE='2022-12-06'`

**Comment by ophub** [comment]
Restore the original system of Android first, and then install armbian

**Comment by D-Sta** [comment]
Unfortunately i did not make a backup of the system, because when i tried doing `amrbian-ddbr` then `b` i used to get an error of not enough space, something like >2GB required, only 1GB available.

Can i use the images provided [Android TV firmware](https://github.com/ophub/kernel/releases/tag/tools)  with the `armbin-ddbr` or those images are only for **Amlogic usb burning tool**

**Comment by ophub** [comment]
Look at the previous reply, there is an Android system download in the post with pictures.

**Comment by D-Sta** [comment]
I have successfully recovered the stock android tv firmware, and from them installed armbian to the emmc. Tried the 5.10.157 kernel by doing `armbian-update -k 5.10.157 -v stable -m yes` but the lan still does not work.

```
[    0.000000] Linux version 5.10.157-ophub (root@ubuntu2204) (Ubuntu clang version 14.0.0-1ubuntu1, Ubuntu LLD 14.0.0) #1 SMP PREEMPT Sun Dec 4 15:30:44 CST 2022
......
[   15.985415] meson8b-dwmac ff3f0000.ethernet eth0: validation of rmii with support 0000000,00000000,00006280 and advertisement 0000000,00000000,00006280 failed: -22
[   15.997961] meson8b-dwmac ff3f0000.ethernet eth0: no phy at addr -1
[   16.003998] meson8b-dwmac ff3f0000.ethernet eth0: stmmac_open: Cannot attach to PHY (error: -19)
```

**Comment by ophub** [comment]
Is the network normal when in USB?

**Comment by luucasmooreira** [comment]
![image](https://user-images.githubusercontent.com/118861204/206751941-e42f73ed-3e2f-4fac-b0de-28e3871dec52.jpeg)


o máximo que consegui foi usar armbian install e usar dtb 520 ( a95xf3-air.dtb). Nao encontro maneira de reconhercer 4GB de Ram e Ip Lan :/

**Comment by D-Sta** [comment]
> Is the kernel network working fine with 5.15.81? Is the network normal when in USB?

Sorry for being late replying to your question, but was doing a lot of testing as to make sure if it works or not. So, the answer is: Yes, it works with kernel 5.15.81, when booting with USB, **and the emmc is in its stock state, armbian not flashed to it yet**. As soon as I flash armbian to the EMMC with the mainline u-boot, the Ethernet no longer works.

But there is a note that might be useful. When restoring the EMMC using the image provided in this issues **SlimboxTV** the Ethernet does not work even when booting from USB. Fortunately I have 2 pcs of the same TV box, so I booted into armbian from USB and did a backup using `armbian-ddbr`. Then used the image to restore the emmc of the first device and then the Ethernet was working with 5.15.81, but only while booting from usb. So I think the problem of Ethernet occurs somewhere in the u-boot.

Now I'm restoring back to stock firmware, and this time will try to install on EMMC but without using the mainline u-boot. As far as I remember everything was working that way, but only 1GB of RAM was available, but need to make sure because I forgot.

**Comment by D-Sta** [comment]
**UPDATE**
@ophub 
I've tried installing 5.10.158/5.4.226 kernel but no luck with the Ethernet. As long as i install the mainline u-boot the ethernet no longer works.

**Comment by ophub** [comment]
1.恢复原厂安装系统
2.下载最新固件
3.安装时加参数启用主线
`armbian-install -m yes`
select `522`

**Comment by D-Sta** [comment]
> 1.恢复原厂安装系统 2.下载最新固件 3.安装时加参数启用主线 `armbian-install -m yes` select `522`

I have already done those steps. But as long as I run do armbian-install -m yes, the ethernet no longer works. But while running from USB, there is no problem.

**Comment by ophub** [comment]
which kernel are you using

**Comment by D-Sta** [comment]
> which kernel are you using

I have tried 5.15/5.10/5.4

**Comment by D-Sta** [comment]
![Front](https://user-images.githubusercontent.com/101727306/206896296-f7dcfde2-69cf-4412-94b0-858a906be7aa.png)
![Back](https://user-images.githubusercontent.com/101727306/206896309-5275c187-1c3f-4d92-b321-3257903a0abe.png)

**Comment by D-Sta** [comment]
This is the dtb that I have extracted from my box
[mybox.gz](https://github.com/ophub/amlogic-s9xxx-armbian/files/10201859/mybox.gz)

**Comment by D-Sta** [comment]
I have used this way to extract the dtb
https://github.com/ophub/amlogic-s9xxx-armbian/tree/main/build-armbian/documents#1213-how-to-decompile-a-dtb-file

Should I try the method you have just provided?

**Comment by D-Sta** [comment]
[mybox.gz](https://github.com/ophub/amlogic-s9xxx-armbian/files/10202151/mybox.gz)

**Comment by luucasmooreira** [comment]
Yes it works. Congratulations to @tv4you2016
